PostHog/Autocapture/PostHogAutocaptureEventTracker.swift

```diff
@@ -0,0 +1,533 @@
+//
+//  PostHogAutocaptureEventTracker.swift
+//  PostHog
+//
+//  Created by Yiannis Josephides on 14/10/2024.
+//
+
+#if os(iOS) || targetEnvironment(macCatalyst)
+    import UIKit
+
+    // TODO: ph-no-capture
+    class PostHogAutocaptureEventTracker {
+        struct EventData {
+            struct ViewNode: CustomStringConvertible {
+                let text: String
+                let targetClass: String
+                let index: Int
+                let subviewCount: Int
+
+                // used when building $elements_chain
+                // Note: For some reason text will not be processed if not present in elements_chain string.
+                // Couldn't pinpoint to exact place in `posthog` repo where we check for this though
+                var description: String {
+                    "\(targetClass)\(text.isEmpty ? "" : ":text=\"\(text)\"")"
+                }
+            }
+
+            enum EventSource {
+                case notification(name: String)
+                case actionMethod(description: String)
+                case gestureRecognizer(description: String)
+            }
+
+            let touchCoordinates: CGPoint?
+            let value: String?
+            let screenName: String?
+            let viewHierarchy: [ViewNode]
+            let targetClass: String
+            let accessibilityLabel: String?
+            let accessibilityIdentifier: String?
+            // values >0 means that this event will be debounced for `debounceInterval`
+            let debounceInterval: TimeInterval
+        }
+
+        static var eventProcessor: (any AutocaptureEventProcessing)? {
```

Comment:
Could this be just a method instead of a static instance class?

Reply:
this commit right? [`8b1406f` (#224)](https://github.com/PostHog/posthog-ios/pull/224/commits/8b1406fc7e2bb378bc9d5facaa469596406ad21c)
I meant something that doesn't need a weak ref and static ref
just a class method that processes what's needed.
having more weak refs and static vars would make the refactoring for multiple instances later on harder.

Reply:
The 2nd point feels something closer to what I thought.
On pto now but feel free to get this ball rolling and dont get blocked, we can always improve later if needed :) so ✅